src/neptune_tensorboard/integration/tensorflow_integration.py

```diff
@@ -13,193 +13,63 @@
-
-    @staticmethod
-    def _calculate_x_value(global_step):
-        if global_step is not None:
-            return int(global_step)
-        else:
-            return time.time()
-
-
-def _integrate_with_tensorflow(experiment_getter, prefix=False):
-    tensorflow_integrator = TensorflowIntegrator(experiment_getter=experiment_getter, prefix=prefix)
-
+def _integrate_with_tensorflow(run, base_namespace):
     try:
         version = "<unknown>"
 
         # noinspection PyUnresolvedReferences
         version = parse_version(tf.version.VERSION)
 
         if version >= parse_version("2.0.0-rc0"):
-            return _patch_tensorflow_2x(experiment_getter, prefix)
-        elif version >= parse_version("1.0.0"):
-            return _patch_tensorflow_1x(tensorflow_integrator)
-
+            return _patch_tensorflow_2x(run, base_namespace)
     except AttributeError:
-        message = (
-            "Unrecognized tensorflow version: {}. Please consider "
-            "upgrading your neptune-client and neptune-tensorboard libraries"
-        )
+        message = "Unrecognized tensorflow version: {}. Please make sure " "that the tensorflow version is >=2.0"
         raise Exception(message.format(version))
 
 
-def _patch_tensorflow_1x(tensorflow_integrator):
-    _add_summary_method = tf.summary.FileWriter.add_summary
-
-    def _neptune_add_summary(summary_writer, summary, global_step=None):
-        tensorflow_integrator.add_summary(summary_writer, summary, global_step)
-        _add_summary_method(summary_writer, summary, global_step=global_step)
-
-    tf.summary.FileWriter.add_summary = _neptune_add_summary
-    tf.summary.FileWriter._original_no_neptune_add_summary = _add_summary_method
-
-    return tensorflow_integrator
-
-
-def _patch_tensorflow_2x(experiment_getter, prefix):
-    try:
-        from tensorflow.python.ops import summary_ops_v2
-    except ImportError:
-        # support TF<2.2
-        from tensorflow_core.python.ops import summary_ops_v2
-
+def _patch_tensorflow_2x(run, base_namespace):
     def get_channel_name(name):
-        if prefix:
-            context = summary_ops_v2.context.context()
-            if hasattr(context, "summary_writer"):
-                writer = context.summary_writer
-            elif "tensorflow.python.ops.summary_ops_v2" in sys.modules:
-                writer = sys.modules["tensorflow.python.ops.summary_ops_v2"]._summary_state.writer
-            else:
-                writer = None
-            if writer is not None:
-                log = str(writer._init_op_fn.keywords["logdir"]._numpy(), "utf-8")
-                writer_name = os.path.split(log)[1]
-                if writer_name:
-                    return writer_name + "_" + name
         return name
 
     _scalar = tf.summary.scalar
     _text = tf.summary.text
     _image = tf.summary.image
+    _graph = tf.summary.graph
 
     def scalar(name, data, step=None, description=None):
```

Comment:
I don't know good reasons for using the inner functions in general. Let's move all these functions to global scope.

Reply:
These are nested function so that we can capture the original tf functions `_scalar = tf.summary.scalar` and dispatch to them.

Reply:
I know but that's just a hacky way of doing this :wink:. You're depending on function access scopes and overriding the internal function metadata (like `__doc__` etc.). Let me try to prepare some snippet to refactor it :wink:

Reply:
Have used `functools.wraps` to take care of this. Let me know if that makes sense. Thanks!

Reply:
👍 Almost :wink:. I've prepared my "point of view": https://github.com/neptune-ai/neptune-tensorboard/pull/53. Let me know how does it fits you :wink: